internal/controller/services/monitoring/resources/prometheus-restricted.tmpl.yaml

```diff
@@ -0,0 +1,220 @@
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: prometheus-restricted
+  namespace: {{.Namespace}}
+  labels:
+    app: prometheus-restricted
+    platform.opendatahub.io/part-of: monitoring
+  ownerReferences:
+  - apiVersion: {{.Component.APIVersion}}
+    kind: {{.Component.Kind}}
+    name: {{.Component.Name}}
+    uid: {{.Component.UID}}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: prometheus-restricted
+  labels:
+    platform.opendatahub.io/part-of: monitoring
+  {{- if .DSCI }}
+  ownerReferences:
+  - apiVersion: {{.DSCI.APIVersion}}
+    kind: {{.DSCI.Kind}}
+    name: {{.DSCI.Name}}
+    uid: {{.DSCI.UID}}
+  {{- end }}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: odh:prometheus-metrics-reader
+subjects:
+  - kind: ServiceAccount
+    name: prometheus-restricted
+    namespace: {{.Namespace}}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: prometheus-restricted-auth-delegator
+  labels:
+    platform.opendatahub.io/part-of: monitoring
+  {{- if .DSCI }}
+  ownerReferences:
+  - apiVersion: {{.DSCI.APIVersion}}
+    kind: {{.DSCI.Kind}}
+    name: {{.DSCI.Name}}
+    uid: {{.DSCI.UID}}
+  {{- end }}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: system:auth-delegator
+subjects:
+  - kind: ServiceAccount
+    name: prometheus-restricted
+    namespace: {{.Namespace}}
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: prometheus-restricted-config
+  namespace: {{.Namespace}}
+  labels:
+    app: prometheus-restricted
+    platform.opendatahub.io/part-of: monitoring
+  ownerReferences:
+  - apiVersion: {{.Component.APIVersion}}
+    kind: {{.Component.Kind}}
+    name: {{.Component.Name}}
+    uid: {{.Component.UID}}
+data:
+  kube-rbac-proxy.yaml: |
+    authorization:
+      rewrites:
+        byQueryParameter:
+          name: "namespace"
+      resourceAttributes:
+        group: ""
+        resource: services
+        subresource: proxy
+        namespace: "{{ "{{ .Value }}" }}"
+        name: data-science-monitoringstack-prometheus
+        verb: get
```

Comment:
Based on the jira I thought

`Using a combination of kube-rbac-proxy and prom-label-proxy, restrict access to users having PodMetrics access`

that we want to restrict it based on PodMetrics access.